hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/store/contracts/precompile/impl/TransferPrecompile.java

```diff
@@ -678,4 +688,138 @@ private AccountAmount aaWith(final AccountID account, final long amount, final b
     public void setCanFallbackToApprovals(final boolean canFallbackToApprovals) {
         this.canFallbackToApprovals = canFallbackToApprovals;
     }
+
+    /**
+     * Given a {@link BalanceChange} that was switched from key-based authorization to approval-based authorization,
+     * tries to update the matching adjustment in the transfer list of the in-progress synthetic {@code CryptoTransfer}
+     * op to reflect the switch.
+     *
+     * <p><b>Important:</b> this method acts via side effects, mutating the given {@code opBuilder} in-place.
+     *
+     * @param opBuilder the builder for the in-progress synthetic {@code CryptoTransfer} op
+     * @param switchedChange the {@code BalanceChange} switched to approval-based authorization
+     */
+    @VisibleForTesting
+    static void updateSynthOpForAutoApprovalOf(
+            @NonNull final CryptoTransferTransactionBody.Builder opBuilder,
+            @NonNull final BalanceChange switchedChange) {
+        if (switchedChange.isForCustomFee()) {
+            // The synthetic CryptoTransfer op won't have any transfer list entry for a custom fee adjustment,
+            // so we can't externalize any more info in this case (given the current record creation logic)
+            return;
+        }
+        if (switchedChange.isForHbar()) {
+            updateSynthOpForAutoApprovalOfHbar(opBuilder, switchedChange);
+        } else {
+            updateSynthOpForAutoApprovalOfToken(opBuilder, switchedChange);
+        }
+    }
+
+    /**
+     * Given an hbar {@link BalanceChange} that was switched from key-based authorization to approval-based
+     * authorization, tries to update the matching adjustment in the hbar transfer list of the in-progress
+     * synthetic {@code CryptoTransfer} op to reflect the switch.
+     *
+     * @param opBuilder the builder for the in-progress synthetic {@code CryptoTransfer} op
+     * @param switchedChange the hbar {@code BalanceChange} switched to approval-based authorization
+     */
+    private static void updateSynthOpForAutoApprovalOfHbar(
+            @NonNull final CryptoTransferTransactionBody.Builder opBuilder,
+            @NonNull final BalanceChange switchedChange) {
+        final var transfersBuilder = opBuilder.getTransfersBuilder();
+        for (int i = 0, n = transfersBuilder.getAccountAmountsCount(); i < n; i++) {
+            final var hbarAdjust = transfersBuilder.getAccountAmountsBuilder(i);
+            if (hbarAdjust.getAccountID().equals(switchedChange.accountId())) {
```

Comment:
@tinker-michaelj should we have a check for debit inside this functions to be clear it's only for debits?
For FT amount also
Not sure how to capture debit for NFTs though.

Reply:
Added debit check for hbar and fungible `BalanceChange`'s.